File: rp/segment.cpp
#include <iostream>
#include <string>
#include <syscall.h>
using namespace std;

// 1st argument: input file
// 2nd argument: output file
// 3rd argument: segment size
int main(int argc, char const *argv[]) {

    // split -b [arg_3] -d [arg_1] segment_
    // ls segment_* | shuf | xargs -I {} sh -c 'cat {}; echo' > [arg_2]
    string command = "split -b ";
    string command2 = "ls segment_* | shuf | xargs -I {} sh -c 'cat {}; echo' >";

    command.append(argv[3]).append(" -d ").append(argv[1]).append(" segment_");
    command2.append(argv[2]);

    system(command.c_str());
    system(command2.c_str());
    system("rm segment_*");
    return 0;
}

File: rp/segment.py
"""used for calling commands and typing of method calls"""
import subprocess
from typing import List

class Segmenter():
    """Handles segmentation of the original file and loading segment chunks"""
    def __init__(self, input_file, output_file, segment_size):
        self.input_file = input_file
        self.output_file = output_file
        self.segment_size = segment_size

    def segment(self) -> None:
        """Segments input file into chunks of segment_size and saves them into the output file"""
        subprocess.call("split -b" + str(self.segment_size) 
                        + " -d " + str(self.input_file) + " segment_", shell=True)
        subprocess.call("ls segment_* | shuf | xargs -I {} sh -c 'cat {}; echo' >"
                        + str(self.output_file), shell=True)
        subprocess.call("rm segment_*", shell=True)

    def get_chunks(self) -> List[str]:
        """Return chunks in output file"""
        result = []

        with open(self.output_file, encoding="ascii") as f:
            for line in f:
                if len(line) > 1:
                    result.append(line[:-1])
                else:
                    result.append(line)
        return result

#def main():
#    seg = Segmenter("in.txt", "chunk_file.txt", 16)
#    seg.segment()
#    print(seg.get_chunks())

#if __name__ == "__main__":
#    main()
